Source/Core/Core/AchievementManager.cpp

```diff
@@ -239,6 +239,37 @@ void AchievementManager::DoFrame()
   }
 }
 
+bool AchievementManager::CanPause()
+{
+  u32 frames_to_next_pause = 0;
+  bool can_pause = rc_client_can_pause(m_client, &frames_to_next_pause);
+  if (!can_pause)
+  {
+    OSD::AddMessage("Cannot spam pausing in hardcore mode.", OSD::Duration::VERY_LONG,
+                    OSD::Color::RED);
+    OSD::AddMessage(
+        fmt::format("Can pause in {} seconds.", static_cast<float>(frames_to_next_pause) / 60),
+        OSD::Duration::VERY_LONG, OSD::Color::RED);
+  }
+  return can_pause;
+}
+
+void AchievementManager::DoIdle()
+{
+  if (!IsGameLoaded())
+    return;
+
+  m_idle_thread = new std::thread([]() {
+    auto* client = AchievementManager::GetInstance().m_client;
+    auto* system = AchievementManager::GetInstance().m_system;
+    while (Core::GetState(*system) == Core::State::Paused)
+    {
+      rc_client_idle(client);
+      Common::SleepCurrentThread(1000);
```

Comment:
This is bad for battery life on mobile platforms, since there's nothing preventing the loop from continuing when the screen is off or another app is in the foreground. While there's no way of enabling RetroAchievements in the Android GUI right now, if you want to add a loop like this, I request that you add an explicit check to not activate RetroAchivements when `ANDROID` is defined.

Reply:
Once https://github.com/JosJuice/dolphin/commit/adc0e051ae64d6c617a12b93f44b9dc1e003400a is merged I'll be using that to only run the idle loop when the app is foregrounded.

Reply:
I'm not planning to create any separate PR for that commit, as it has no use on its own. Please cherry-pick it either into this PR or into a later follow-up PR.

Reply:
And for clarity, if you want to address the Android problem in a follow-up PR, please explicitly disable RetroAchivements on Android for now.